Userland/Libraries/LibWeb/HTML/HTMLInputElement.cpp

```diff
@@ -85,7 +85,12 @@ void HTMLInputElement::set_checked(bool checked, ChangeSource change_source)
         m_dirty_checkedness = true;
 
     m_checked = checked;
-    set_needs_style_update(true);
+
+    // This element's :checked pseudo-class could be used in a sibling's sibling-selector,
+    // so we need to invalidate the style of all siblings.
+    parent()->for_each_child([&](auto& child) {
+        child.invalidate_style();
+    });
```

Comment:
Can we just say `invalidate_style()` here? (Invalidation is inclusively recursive.)

Reply:
Oh wait, duh! You're invalidating the *siblings* obviously 😅

Reply:
We could, it just isn't strictly necessary to invalidate parents own styles. Anyway, I'll change it, I was questioning the necessity of preciseness myself.

Reply:
Okay, now that you merged I won't 😜

Reply:
ah now I'm reading your comment. I thought you were suggesting to say `parent.invalidate_style()`